docs/DEPLOYMENT.md

```diff
@@ -0,0 +1,206 @@
+# Deploying EVE-OS
+
+Deploying EVE-OS is similar to deploying any regular operating system. This can happen
+in either physical (EVE-OS runs on bare metal) or virtual (EVE-OS runs inside of a
+virtual machine) environments. When running in a virtual environment, EVE-OS needs to
+have access to [nested virtualization](https://en.wikipedia.org/wiki/Virtualization#Nested_virtualization).
+While it is possible to run EVE-OS in a virtual environment without nested virtualization
+enabled a lot of capabilities (e.g. running accelerated VMs) will be degraded.
+
+## Live vs installer EVE-OS images
+
+Since EVE-OS follows a very traditional boot process, its *live* image has to be
+available on a storage device that is accessible by the BIOS. When it is already
+deployed EVE-OS provides capabilities for managing its own live image. However,
+in order to bootstrap EVE-OS the initial live image has to be written to the
+storage device somehow.
+
+In a lot of situations, the storage device can be written to separately and then
+attached back to the edge node. This applies to both physical (e.g.  taking a live
+EVE-OS image, writing it onto a USB stick and booting off of that USB stick) and
+virtual (taking a live EVE-OS image and instructing your virtualization platform
```

Comment:
This reads as if it is recommended to clone existing EVE images, but that has issues since you end up with a clone of the /config partition, hence identical device.cert.pem, soft-serial, and if no TPM device.key.pem.
If there is no TPM this become a clone, which will be confusing (the controller will see info and metrics messages from two devices as one device - last message wins); FWIW I've done this accidentally running qemu in different workspaces.
If there is a TPM on the device I do not know what happens; EVE-OS will think it is onboarded since there is a /config/device.cert.pem, but there is no keypair in the TPM. I suspect it will crash and burn.

So I don't think we need to either say this isn't supported, or we need to specify which files in /config MUST be deleted as part of the image cloning operation. And we might also want to add some code to EVE-OS to check that a device.cert.pem is consistent with that is in the TPM.

Reply:
I found this whole section somewhat confusing. Can we start with, "there are two versions of eve images: live and installer. Here are the differences, here is when you would choose one over the other, and here is how you do it"?

Reply:
Opted out to clarifying as per @eriknordmark suggestions. @deitch -- feel free to send PRs with better working, positioning.